### DISCUSSION TITLE
"+ Add Subnet" under the "Configure --> Externals" section not doing anything

### DISCUSSION THREAD

**Discussion by ashmatash76** [Azure/ipam#316] [Q&A] [answered]
"+add subnet" doesn't appear to be working.
I was able to add an external network without any issues, but the option to add a subnet within that network doesn't appear to be doing anything. I click it and nothing happens. I tried with 2 different browsers (Chrome and Edge) and got the same behavior.

<img width="1906" alt="2024-08-30_12-42-21" src="https://github.com/user-attachments/assets/b76ac3a8-91c1-434c-8bd9-042bbe62d7ec">

**Comment by DCMattyG** [comment]
Hey @ashmatash76, can you help me understand better how you have the Azure IPAM service deployed? Is this deployed into a Private Container Registry or did you just do a basic deployment leveraging the Public ACR which we host?

**Reply by ashmatash76** [reply]
Its deployed in a Private ACR

**Reply by DCMattyG** [reply]
Can you navigate to this URL:

https://&lt;your app url&gt;/api/status

What version are you running?

**Reply by ashmatash76** [reply]
{
  "status": "OK",
  "version": "3.3.0",
  "stack": "AppContainer",
  "environment": "AZURE_PUBLIC",
  "container": {
    "image_id": "debian",
    "image_version": "12",
    "image_codename": "bookworm",
    "image_pretty_name": "Debian GNU/Linux 12 (bookworm)"
  }
}

**Reply by DCMattyG** [reply]
Interesting. I'm not seeing this behavior on my end as I know this was a known issue in 3.2.0 and I addressed it in 3.3.0.

When was this environment deployed? Recently, or a while back?

**Reply by ashmatash76** [reply]
The tenant subscription has been in production for a few years.

**Reply by DCMattyG** [reply]
Understood @ashmatash76, v3.4.0 will be done very soon so I'll have you test that code out as soon as it drop and I think we'll have this resolved for you.

Standby...

**Reply by ashmatash76** [reply]
Thanks! I'll be on the lookout for the new version.

**Reply by DCMattyG** [reply]
FYI @ashmatash76, v3.4.0 is out so please upgrade to the latest release when you get a chance and let me know how everything is working.

If you're using the standard deployment, simply restart your App Service so it can pull down the latest containers.

Hope that fixes everything for you!

**Comment by ashmatash76** [comment]
Thank you so much for all the help here.
Restarted the application service but /api/status is still showing "version":"3.3.0"
Is this because I deployed with a Private ACR?

**Reply by DCMattyG** [reply]
Ah yes @ashmatash76, there are a few extra steps if you've deployed using a Private ACR.

There is an update script in the **deploy** folder, and you can use it like this:

1. Use `git pull` to fetch the latest version of the Azure IPAM repo, or download the latest version from GitHub
2. Verify you are on the **main** branch by using `git branch`
3. Switch to the `deploy` folder
4. Verify that you are logged into the Azure PowerShell SDK and your context is set to the correct Subscription
5. Verify that you are logged into the Azure CLI and your context is set to the correct Subscription
6. Run `.\update.ps1 -AppName <name of your App Service> -ResourceGroupName <name of the RG your App Service is in>`

The update script will need to be run from an end point that can communicate to the target Azure App Service where IPAM is running. The script will take care of building the new containers, pushing them to the ACR, and restarting the App Service.

Please give that a try and let me know how things are looking after the update. Thanks so much!

**Reply by ashmatash76** [reply]
I managed to figure out how to update GIT and run the update, but that button still doesn't do anything :(

**Reply by DCMattyG** [reply]
Is the `/api/status` URL now showing v3.4.0?

**Comment by ashmatash76** [comment]
Yes, it is showing 3.4.0 now.
________________________________

**Comment by ashmatash76** [comment]
I'm trying to redeploy from scratch. I'll confirm and update with results.
Thanks again!


________________________________

**Comment by ashmatash76** [comment, marked as answer]
I redeployed 3.3.4 from scratch. It appears to be working now.
Thanks for the help!!

________________________________

**Reply by DCMattyG** [reply]
I'm SO GLAD to hear that! Thank you for your patience working through this with me.

Please let me know if anything else comes up, we're here to help!
